packages/h5wasm/create_h5_sample.py

```diff
@@ -0,0 +1,248 @@
+    add_scalar(h5, "ascii_vlen", b"Some text")
+    add_scalar(
+        h5,
+        "ascii_fixed",
+        np.string_("Some text"),
+    )
+    add_scalar(h5, "utf8_vlen", "Some text")
+    add_array(h5, "utf8_vlen", np.array(["foo", "bar", "baz"], h5py.string_dtype()))
+    add_scalar(h5, "utf8_fixed", "Some text", h5py.string_dtype("utf-8", 9))
+
+    # === H5T_BITFIELD ===
+
+    # h5py syntax unknown
+
+    # === H5T_OPAQUE ===
+
+    add_scalar(h5, "byte_string", np.void(b"\x00\x11\x22"))
+    add_scalar(h5, "datetime64", np.void(np.datetime64("2019-09-22T17:38:30")))
+    add_scalar(h5, "datetime64_not-a-time", np.void(np.datetime64("NaT")))
+
+    # === H5T_COMPOUND (and H5T_ARRAY) ===
+
+    add_scalar(
+        h5,
+        "complex64",
+        np.complex64(
+            np.finfo(np.complex64).smallest_normal + np.finfo(np.complex64).max * 1j
+        ),
+    )
+    add_array(
+        h5, "complex64", np.array([[1 + 2j, 3 + 4j], [5 + 6j, 7 + 8j]], np.complex64)
+    ),
+    add_scalar(
+        h5,
+        "complex128",
+        np.complex128(
+            np.finfo(np.complex128).smallest_normal + np.finfo(np.complex128).max * 1j
+        ),
+    )
+    add_scalar(h5, "complex128_BE", 1 + 2j, np.dtype(">c16"))
+    add_array(
+        h5, "complex128", np.array([[1 + 2j, 3 + 4j], [5 + 6j, 7 + 8j]], np.complex128)
+    ),
+    add_scalar(
+        h5,
+        "complex256",
+        np.complex256(
+            np.finfo(np.complex256).smallest_normal + np.finfo(np.complex256).max * 1j
+        ),
+    )
+    add_array(
+        h5, "complex256", np.array([[1 + 2j, 3 + 4j], [5 + 6j, 7 + 8j]], np.complex256)
+    ),
+
+    add_scalar(
+        h5,
+        "compound",
+        (1, 2.0, "foo"),
+        [("bigint", np.int64), ("double", np.float64), ("utf-8", h5py.string_dtype())],
+    )
+    for_ref = add_array(
+        h5,
+        "compound",
+        np.array(
+            [(1, np.nan, "foo"), (2, np.inf, "bar"), (3, np.NZERO, "baz")],
+            [
+                ("bigint", np.int64),
+                ("double", np.float64),
+                ("utf-8", h5py.string_dtype()),
+            ],
+        ),
+    )
+    add_scalar(
+        h5,
+        "compound_nested",
+        ((True, 1 + 2j),),
+        [("nested", [("bool", np.bool_), ("cplx", np.complex64)])],
+    )
+
+    comp = add_array(
+        h5,
+        "compound_array_vlen",
+        np.array(
+            [
+                (
+                    np.array([0, 1], np.float32),
+                    np.array([0], np.uint64),
+                ),
+                (
+                    np.array([2, 3], np.float32),
+                    np.array([0, 1], np.uint64),
+                ),
+                (
+                    np.array([4, 5], np.float32),
+                    np.array([0, 1, 2], np.uint64),
+                ),
+            ],
+            [("arr", np.float32, (2,)), ("vlen", h5py.vlen_dtype(np.uint64))],
+        ),
+    )
```

Comment:
I've managed to unlock the correct syntax for creating a compound with an `H5T_ARRAY` and an `H5T_VLEN` field. :tada: